frontend/src/lib/components/widgets/ChatInput/ChatInput.tsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  useEffect,
+  useRef,
+  useState,
+  ChangeEvent,
+  KeyboardEvent,
+} from "react"
+import { useTheme } from "@emotion/react"
+import { Send } from "@emotion-icons/material-rounded"
+import { Textarea as UITextArea } from "baseui/textarea"
+
+import { ChatInput as ChatInputProto } from "src/lib/proto"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
+import Icon from "src/lib/components/shared/Icon"
+import InputInstructions from "src/lib/components/shared/InputInstructions/InputInstructions"
+
+import {
+  StyledChatInputContainer,
+  StyledChatInput,
+  StyledSendIconButton,
+} from "./styled-components"
+
+const MIN_HEIGHT = 40.4 // 40.4 is the default height of a text input
+const MAX_HEIGHT = 230
+
+export interface Props {
+  disabled: boolean
+  element: ChatInputProto
+  widgetMgr: WidgetStateManager
+  width: number
+}
+
+const isEnterKeyPressed = (
+  event: KeyboardEvent<HTMLTextAreaElement>
+): boolean => {
+  // Using keyCode as well due to some different behaviors on Windows
+  // https://bugs.chromium.org/p/chromium/issues/detail?id=79407
+
+  const { keyCode, key } = event
+  return key === "Enter" || keyCode === 13 || keyCode === 10
+}
+
+function ChatInput({ width, element, widgetMgr }: Props): React.ReactElement {
+  const theme = useTheme()
+  // True if the user-specified state.value has not yet been synced to the WidgetStateManager.
+  const [dirty, setDirty] = useState(false)
+  // The value specified by the user via the UI. If the user didn't touch this widget's UI, the default value is used.
+  const [value, setValue] = useState("")
+  // The value of the height of the textarea. It depends on a variety of factors including the default height, and autogrowing
+  const [scrollHeight, setScrollHeight] = useState(0)
+  const chatInputRef = useRef<HTMLTextAreaElement>(null)
+
+  const getScrollHeight = (): number => {
+    let scrollHeight = 0
+    const { current: textarea } = chatInputRef
+    if (textarea) {
+      textarea.style.height = "auto"
+      scrollHeight = textarea.scrollHeight
+      textarea.style.height = ""
+    }
+
+    return scrollHeight
+  }
+
+  const handleSubmit = (): void => {
+    widgetMgr.setStringTriggerValue(element, value, { fromUi: true })
```

Comment:
I think we need to add a check here to only trigger the submit if the value actually contains something, e.g.:

```suggestion
    if (!value) {
       return
    }
    widgetMgr.setStringTriggerValue(element, value, { fromUi: true })
```